extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryManager.java

```diff
@@ -0,0 +1,241 @@
+import org.apache.rya.streams.querymanager.QueryExecutor.QueryExecutorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.AbstractScheduledService.Scheduler;
+
+import edu.umd.cs.findbugs.annotations.DefaultAnnotation;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A service for managing {@link StreamsQuery} running on a Rya Streams system.
+ * <p>
+ * Only one QueryManager needs to be running to manage any number of rya
+ * instances/rya streams instances.
+ */
+@DefaultAnnotation(NonNull.class)
+public class QueryManager extends AbstractIdleService {
+    private static final Logger LOG = LoggerFactory.getLogger(QueryManager.class);
+
+    private final String ryaInstanceName;
+    private final QueryExecutor queryExecutor;
+    private final Scheduler scheduler;
+    private final Set<QueryRepository> queryRepos;
+
+    private final Map<String, QueryChangeLogSource> sources;
+    private final Map<UUID, StreamsQuery> queriesCache;
+
+    /**
+     * Creates a new {@link QueryManager}.
+     *
+     * @param ryaInstanceName - The Rya Instance to connect to. (not null)
+     * @param queryExecutor - The {@link QueryExecutor} that starts/creates
+     *        queries when a CREATED query is found in the
+     *        {@link QueryChangeLog}. (not null)
+     * @param sources - The {@link QueryChangeLogSource}s the sources of
+     *        QueryChangeLogs. (not null)
+     * @param queryRepos - The set of {@link QueryRepository}s that will notify
+     *        this manager when {@link ChangeLogEntry}s occur. (not null)
+     * @param scheduler - The {@link Scheduler} to use throughout
+     */
+    public QueryManager(final String ryaInstanceName, final QueryExecutor queryExecutor,
+            final Map<String, QueryChangeLogSource> sources, final Set<QueryRepository> queryRepos,
+            final Scheduler scheduler) {
+        this.ryaInstanceName = requireNonNull(ryaInstanceName);
+        this.sources = requireNonNull(sources);
+        this.queryExecutor = requireNonNull(queryExecutor);
+        this.queryRepos = requireNonNull(queryRepos);
+        this.scheduler = requireNonNull(scheduler);
+
+        // subscribe to the repos and sources to be notified of changes.
+        queryRepos.forEach(repo -> repo.subscribe(new QueryManagerQueryChange()));
+        sources.forEach((ryaInstance, source) -> source.subscribe(new QueryManagerSourceListener()));
+
+        // create query cache.
+        queriesCache = new HashMap<>();
+    }
+
+    /**
+     * Starts running a query.
+     *
+     * @param query - The query to run.(not null)
+     */
+    private void runQuery(final StreamsQuery query) {
+        requireNonNull(query);
+        LOG.trace("Starting Query: " + query.getSparql());
+        if (!queryExecutor.isRunning()) {
```

Comment:
This case will never happen since you start it in this service's startup method.